contracts/interfaces/IStakedTokenDataProvider.sol

```diff
@@ -59,83 +54,67 @@ interface IStakedTokenDataProvider {
     external
     view
-    returns (
-      StakedTokenData memory stkAaveData,
-      StakedTokenData memory stkBptData,
-      uint256 ethPrice
-    );
+    returns (StakedTokenData memory stakedAssetData);
 
   /**
-   * @notice Returns data of Staked Aave
-   * @return stkAaveData An object with StkAave data
+   * @notice Retrieves staked token data and user-specific data for a given user, staked asset, and its associated oracle.
+   * @param user The address of the user for whom the data is to be fetched.
+   * @param stakedAsset The address representing the staked token.
+   * @param oracleAddress The address representing the oracle associated with the staked token.
+   * @return stakedTokenData `StakedTokenData` struct with details about the staked asset.
+   * @return stakedUserData `StakedTokenUserData` struct containing user-specific details related to the staked asset.
    */
-  function getStkAaveData() external view returns (StakedTokenData memory stkAaveData);
-
-  /**
-   * @notice Returns data of Staked Bpt Aave
-   * @return stkBptData An object with StkBpt data
-   */
-  function getStkBptData() external view returns (StakedTokenData memory stkBptData);
-
-  /**
-   * @notice Returns user data of all Staked Tokens
-   * @param user The address of the user
-   * @return stkAaveData An object with StkAave data
-   * @return stkAaveUserData An object with user data of StkAave
-   * @return stkBptData An object with StkBpt data
-   * @return stkBptUserData An object with user data of StkBpt
-   * @return ethPrice The price of ETH, USD denominated (expressed with 8 decimals)
-   */
-  function getAllStakedTokenUserData(address user)
+  function getStakedUserData(
+    address user,
+    address stakedAsset,
+    address oracleAddress
+  )
     external
     view
-    returns (
-      StakedTokenData memory stkAaveData,
-      StakedTokenUserData memory stkAaveUserData,
-      StakedTokenData memory stkBptData,
-      StakedTokenUserData memory stkBptUserData,
-      uint256 ethPrice
-    );
+    returns (StakedTokenData memory stakedTokenData, StakedTokenUserData memory stakedUserData);
 
   /**
-   * @notice Returns user data of Staked Aave
-   * @param user The address of the user
-   * @return stkAaveData An object with StkAave data
-   * @return stkAaveUserData An object with user data of StkAave
+   * @notice Retrieves data for a batch of staked assets given their addresses and the addresses of their associated oracles.
+   * @param stakedTokens An array of addresses representing the staked tokens.
+   * @param oracleAddresses An array of addresses representing the oracles associated with the staked tokens.
+   * @return stakedData An array containing data about the staked assets.
+   * @return prices An array containing the latest price data from the oracles for the staked tokens.
+   * @return ethPrice An array containing the latest ETH price for the staked tokens.
```

Comment:
does not need to be an array, it's just the ETH/USD price, same for all staked tokens